pkg/apis/networkoperator/v1/types.go

```diff
@@ -119,6 +123,59 @@ type OpenshiftSDNConfig struct {
 	UseExternalOpenvswitch *bool `json:"useExternalOpenvswitch,omitempty"`
 }
 
+type OpenStackCredentials struct {
+	// TODO(dulek): Might want to add all the options, e.g. keys.
+	AuthURL           string `json:"authURL"`
+	UserDomainName    string `json:"userDomainName"`
+	Username          string `json:"username"`
+	Password          string `json:"password"`
+	ProjectDomainName string `json:"projectDomainName"`
+	ProjectId         string `json:"projectId"`
+}
+
+// KuryrConfig configures the Kuryr-Kubernetes SDN
+type KuryrConfig struct {
+	// The base64-encoded CA certificate to use when connecting to OpenStack.
+	// +optional
+	OpenStackCACertificate string `json:"openStackCACertificate"`
+
+	OpenStackCredentials *OpenStackCredentials `json:"openStackCredentials"`
+
+	// If true, kuryr-daemon readiness and liveness probes are enabled.
+	// +optional
+	DaemonEnableProbes *bool `json:"daemonEnableProbes"`
+
+	// The port kuryr-daemon will listen for readiness and liveness requests.
+	// +optional
+	DaemonProbesPort *uint16 `json:"daemonProbesPort"`
+
+	// The CNI bin directory on the hosts, usually /opt/cni/bin.
+	CNIBinDir string `json:"cniBinDir"`
+
+	// If true, kuryr-controller readiness and liveness probes are enabled.
+	// +optional
+	ControllerEnableProbes *bool `json:"controllerEnableProbes"`
+
+	// The port kuryr-controller will listen for readiness and liveness requests.
+	// +optional
+	ControllerProbesPort *uint16 `json:"controllerProbesPort"`
+
+	// TODO(dulek): Might want to make that a list (scope?)
+	PodSecurityGroups string `json:"podSecurityGroups"`
+
+	PodSubnet string `json:"podSubnet"`
+
+	ServiceSubnet string `json:"serviceSubnet"`
```

Comment:
Is there a reason that the IP addresses can't come from the networks above? I would prefer that. It's OK if Kuryr only supports a single ClusterCIDR.

Reply:
Uh, this one is complicated. So I should be able to fetch required Neutron subnet ID's using serviceNetwork if I'm allowed to connect to cloud provider (OpenStack) from inside the operator.

With clusterNetworks we work totally differently, so I should probably go with the idea you've described and catch multiple in the validation.